phoenix-core/src/it/java/org/apache/phoenix/end2end/salted/SaltedTableVarLengthRowKeyIT.java

```diff
@@ -87,4 +89,36 @@ public void testSelectValueWithPointKeyQuery() throws Exception {
             conn.close();
         }
     }
+
+    @Test
+    public void testSaltedVarbinaryUpperBoundQuery() throws Exception {
+        String tableName = generateUniqueName();
+        String ddl = "CREATE TABLE " + tableName +
+                " ( k VARBINARY PRIMARY KEY, a INTEGER ) SALT_BUCKETS = 3";
+        String dml = "UPSERT INTO " + tableName + " values (?, ?)";
+        String sql2 = "SELECT * FROM " + tableName + " WHERE k = ?";
+
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.createStatement().execute(ddl);
+            PreparedStatement stmt = conn.prepareStatement(dml);
+            stmt.setInt(2, 1);
+
+            stmt.setBytes(1, new byte[] { 5 });
```

Comment:
Any reason why the sequence is different? setting 1st parameter followed by second makes it more readable.

Reply:
NVM. I assume it is because of logic on upsert.